### ISSUE TITLE
Escape bracket gets escaped in messages

### ISSUE THREAD

**Issue by nsdeschenes** [lingui/js-lingui#897]
**Describe the bug**
In linguijs v2, I was able to escape back ticks using `\`, however now in linguijs v3 the escape brackets are being escaped and thus breaking the translations when it tries to go and find the doubly escaped `\\`.

**To Reproduce**
```js
i18n._(t`Passing \`${argSet}\` in \`${fieldSet}\` is not supported.`)
```
Produces:
```
msgid "Passing \\`${argSet}\\` in \\`${fieldSet}\\` is not supported. "
msgstr "Passing \\`${argSet}\\` in \\`${fieldSet}\\` is not supported. "
```

**Expected behavior**
Previously in v2 running extract on the original string would produce
```
msgid "Passing `${argSet}` in `${fieldSet}` is not supported. "
msgstr "Passing `${argSet}` in `${fieldSet}` is not supported. "
```

- jsLingui version ``@lingui/cli@3.2.3``
- Babel version ``@babel/core@7.12.9``
- Your Babel config: Marcos Plugin, and ``@babel/preset-env, targets: node``

**Comment by semoal** [comment]
This could be fixed using the cooked value from babel extractor.
https://exploringjs.com/impatient-js/ch_template-literals.html#cooked-vs.-raw-template-strings-advanced
But not sure if this would be a breaking change or if it makes sense.

Example:

```sh
    Node {
      type: 'TemplateElement',
      start: 54,
      end: 65,
      loc: SourceLocation {
        start: Position { line: 3, column: 10 },
        end: Position { line: 3, column: 21 }
      },
      value: { raw: 'Variable \\`', cooked: 'Variable `' },
      tail: false,
      leadingComments: undefined,
      innerComments: undefined,
      trailingComments: undefined
    }
```

https://github.com/lingui/js-lingui/blob/main/packages/macro/src/macroJs.ts#L267

**Comment by semoal** [comment]
I'm not sure if the pull request i opened is a good solution for this, probably Tomas has other opinion or something differnt in mind

**Comment by tricoder42** [comment]
@semoal I've never heard about `cooked` prop. It's first time I see it. I used `raw` before but that actually didn't work with html entities.

If you feel it would solve the problem in a more elegant way, feel free to use it. Usually regexp is a workaround in situations like this because it doesn't take into account if the backslashes are escaped or not. I guess you would need a stateful parser for a correct colution and that might be what `cooked` value is. Just my thoughts, I have zero knowledge about underlying concepts.

**Comment by semoal** [comment]
😱  Released 3.3.0 with this fix/feature introduced!

```
npx update-by-scope @lingui
```

**Comment by nsdeschenes** [comment]
Hi Guys, I have finally got around to attempting to update to lingui V3 however I have now encountered this. It seems that the fix only finds the first escape `\` and not the others present throughout the rest of the string.

```js
i18n._(t`Passing \`${argSet}\` in \`${fieldSet}\` is not supported.`)
```
Produces:
```
msgstr "Passing `${argSet}\\` in \\`${fieldSet}\\` is not supported. "
```

**Expected behavior**
Previously in v2 running extract on the original string would produce
```
msgid "Passing `${argSet}` in `${fieldSet}` is not supported. "
msgstr "Passing `${argSet}` in `${fieldSet}` is not supported. "
```

- jsLingui version ``@lingui/cli@3.3.0``
- Babel version ``@babel/core@7.12.10``
- Your Babel config: Marcos Plugin, and ``@babel/preset-env, targets: node``

**Comment by semoal** [comment]
https://github.com/lingui/js-lingui/blob/main/packages/macro/src/macroJs.ts#L11 probably in this line the g is required to remove all the occurences, could you try to patch your node_modules and give it a try?

**Comment by nsdeschenes** [comment]
Tried that and it fixed the issues, wondering if this change will be in the next update to the package ?

**Comment by semoal** [comment]
Yes absolutely

**Comment by semoal** [comment]
Released in 3.4.0 🥳
